src/Compiler/TypedTree/TypedTreeOps.fs

```diff
@@ -9691,80 +9693,92 @@ let rec EvalAttribArgExpr (g: TcGlobals) x =
     | TypeOfExpr g _ -> x
     | TypeDefOfExpr g _ -> x
     | Expr.Op (TOp.Coerce, _, [arg], _) -> 
-        EvalAttribArgExpr g arg
+        EvalAttribArgExpr true g arg
     | EnumExpr g arg1 -> 
-        EvalAttribArgExpr g arg1
+        EvalAttribArgExpr true g arg1
     // Detect bitwise or of attribute flags
     | AttribBitwiseOrExpr g (arg1, arg2) -> 
-        let v1 = EvalAttribArgExpr g arg1
+        let v1 = EvalAttribArgExpr false g arg1
 
         match v1 with
         | IntegerConstExpr ->
-            EvalArithBinOp ((|||), (|||), (|||), (|||), (|||), (|||), (|||), (|||), ignore2, ignore2) v1 (EvalAttribArgExpr g arg2) 
+            EvalArithBinOp ((|||), (|||), (|||), (|||), (|||), (|||), (|||), (|||), ignore2, ignore2) v1 (EvalAttribArgExpr false g arg2) 
```

Comment:
Good point, since bitwise OR does currently work to some extent, I can't do the check here, and then this particular case would slip under the radar. I'm afraid making it bullet-proof would make the code a lot messier. Not sure it's worth it for such an edge case, plus the world does not end if someone manages to make use of this in 7.0.

Reply:
I'll enable the check, but things like `(1 + 1) ||| (1 + 1) ||| (1 + 1)` will raise multiple errors. It's probably the better option.

Reply:
Multiple errors is fine in my opinion, since it is multiple sub-expressions